tests/acceptance/features/bootstrap/SpacesTUSContext.php

```diff
@@ -0,0 +1,161 @@
+
+	/**
+	 * @When /^user "([^"]*)" uploads a file from "([^"]*)" to "([^"]*)" via TUS inside of the space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $source
+	 * @param string $spaceName
+	 * @param string $destination
+	 * @param array|null $uploadMetadata
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi(string $user, string $source, string $destination, string $spaceName, ?array $uploadMetadata = null) {
+		$space = $this->spacesContext->getSpaceByName($user, $spaceName);
+		if (!WebDavHelper::$SPACE_ID_FROM_OCIS) {
+			WebDavHelper::$SPACE_ID_FROM_OCIS = $space['id'];
+		}
+		$this->tusContext->userUploadsUsingTusAFileTo($user, $source, $destination);
+		WebDavHelper::$SPACE_ID_FROM_OCIS = null;
+	}
+
+	/**
+	 * @Given user :user has created a new TUS resource for the space :spaceName using the WebDAV API with these headers:
+	 *
+	 * @param string $user
+	 * @param string $spaceName
+	 * @param TableNode $headers
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userHasCreatedANewTusResourceForTheSpaceUsingTheWebdavApiWithTheseHeaders(string $user, string $spaceName, TableNode $headers):void {
+		$this->userCreatesANewTusResourceForTheSpaceUsingTheWebdavApiWithTheseHeaders($user, $spaceName, $headers);
+		$this->featureContext->theHTTPStatusCodeShouldBe(201, "Expected response status code should be 201");
+	}
+
+	/**
+	 * @When user :user creates a new TUS resource for the space :spaceName using the WebDAV API with these headers:
+	 *
+	 * @param string $user
+	 * @param string $spaceName
+	 * @param TableNode $headers
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userCreatesANewTusResourceForTheSpaceUsingTheWebdavApiWithTheseHeaders(string $user, string $spaceName, TableNode $headers):void {
+		$space = $this->spacesContext->getSpaceByName($user, $spaceName);
+		if (!WebDavHelper::$SPACE_ID_FROM_OCIS) {
+			WebDavHelper::$SPACE_ID_FROM_OCIS = $space['id'];
+		}
+		$this->tusContext->createNewTUSResourceWithHeaders($user, $headers, '');
+	}
+
+	/**
+	 * @When /^user "([^"]*)" uploads a file with content "([^"]*)" to "([^"]*)" via TUS inside of the space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $content
+	 * @param string $resource
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 * @throws Exception|GuzzleException
+	 */
+	public function userUploadsAFileWithContentToViaTusInsideOfTheSpaceUsingTheWebdavApi(string $user, string $content, string $resource, string $spaceName) {
+		$space = $this->spacesContext->getSpaceByName($user, $spaceName);
+		if (!WebDavHelper::$SPACE_ID_FROM_OCIS) {
+			WebDavHelper::$SPACE_ID_FROM_OCIS = $space['id'];
+		}
+		$this->tusContext->userUploadsAFileWithContentToUsingTus($user, $content, $resource);
+	}
```

Comment:
```suggestion
	WebDavHelper::$SPACE_ID_FROM_OCIS = '';
	}
```
Is clearing space ID not applicable here?

Reply:
No we don't it's done in OC10